---
title: "Data Processing YRI+human data"
output: html_document
date: '2023-07-06'
---

This code goes through the data processing of GH batch 1 and batch 2 data. 

Batch 2 Data Processing Part 1
```{r}

dyn.load('/software/geos-3.7.0-el7-x86_64/lib/libgeos_c.so.1')
dyn.load('/software/hdf5_hl-1.8.17-el7-x86_64/lib/libhdf5_hl.so.10')

library(dplyr)
library(Seurat)

#Creating a Seurat object
GH_data_batch2 <- '/project2/gilad/emaan/ghousman_data/human_chimp_chondro_time_batch2_h_06/outs/multi/count/raw_feature_bc_matrix'

list.files(GH_data_batch2) 
data_batch2 <- Read10X(data.dir = GH_data_batch2)

#This Seurat object only has the gene expression data
GH_seurat_batch_2 = CreateSeuratObject(counts = data_batch2$`Gene Expression`)

confidence_table_batch_2 <- read.csv("/project2/gilad/emaan/ghousman_data/human_chimp_chondro_time_batch2_h_06/outs/multi/multiplexing_analysis/assignment_confidence_table.csv")
```

Batch 2 Data Processing Part 2
Adding confidence values into the meta data. Cells with a probability less than 0.7 are removed.
```{r}
GH_seurat_batch_2$condition <- ""

cell.idents <- list()
cmo.names <- names(confidence_table_batch_2[,2:7])
for (cmo in cmo.names) {
  cell.idents[[cmo]] <- confidence_table_batch_2$Barcode[which(confidence_table_batch_2[[cmo]]>0.7)]
  print(length(cell.idents[[cmo]]))
}

for (cmo in cmo.names) {
  w <- which(colnames(GH_seurat_batch_2) %in% cell.idents[[cmo]])
GH_seurat_batch_2$condition[w] <- cmo
}

#Changing names from CMO to condition
GH_seurat_batch_2@meta.data$condition[which(GH_seurat_batch_2@meta.data$condition=="CMO301")] <- "H5_D0"
GH_seurat_batch_2@meta.data$condition[which(GH_seurat_batch_2@meta.data$condition=="CMO303")] <- "H5_D14c"
GH_seurat_batch_2@meta.data$condition[which(GH_seurat_batch_2@meta.data$condition=="CMO304")] <- "H3_D7m"
GH_seurat_batch_2@meta.data$condition[which(GH_seurat_batch_2@meta.data$condition=="CMO306")] <- "H3_D14c"
GH_seurat_batch2_df= GH_seurat_batch_2@meta.data

GH_seurat_batch2_clean <- subset(GH_seurat_batch_2, subset = condition != "")
GH_seurat_batch2_clean_df <- GH_seurat_batch2_clean@meta.data

#Checking to make sure that we did not lose any cells in the process. The number of cells assigned to each CMO here should be the same as lines #30-35
value_counts <- table(GH_seurat_batch2_clean_df$condition)
print(value_counts)

save(GH_seurat_batch2_clean, file = "GH_seurat_batch2_clean.RData")

```

Batch 1 Data Processing Part 1
```{r}

dyn.load('/software/geos-3.7.0-el7-x86_64/lib/libgeos_c.so.1')
dyn.load('/software/hdf5_hl-1.8.17-el7-x86_64/lib/libhdf5_hl.so.10')

library(dplyr)
library(Seurat)

#Creating a Seurat object
GH_data_batch1 <- '/project2/gilad/emaan/ghousman_data/human_chimp_chondro_time_batch1_h_06/outs/multi/count/raw_feature_bc_matrix'

list.files(GH_data_batch1) 
data_batch1 <- Read10X(data.dir = GH_data_batch1)

#This Seurat object only has the gene expression data
GH_seurat_batch_1 = CreateSeuratObject(counts = data_batch1$`Gene Expression`)

confidence_table_batch_1 <- read.csv("/project2/gilad/emaan/ghousman_data/human_chimp_chondro_time_batch1_h_06/outs/multi/multiplexing_analysis/assignment_confidence_table.csv")
```

Batch 1 Data Processing Part 2
Adding confidence values into the meta data. Cells with a probability less than 0.7 are removed.
```{r}
GH_seurat_batch_1$condition <- ""

cell.idents <- list()
cmo.names <- names(confidence_table_batch_1[,2:13])
for (cmo in cmo.names) {
  cell.idents[[cmo]] <- confidence_table_batch_1$Barcode[which(confidence_table_batch_1[[cmo]]>0.7)]
  print(length(cell.idents[[cmo]]))
}

for (cmo in cmo.names) {
  w <- which(colnames(GH_seurat_batch_1) %in% cell.idents[[cmo]])
GH_seurat_batch_1$condition[w] <- cmo
}

#Changing names from CMO to condition
GH_seurat_batch_1@meta.data$condition[which(GH_seurat_batch_1@meta.data$condition=="CMO301")] <- "H1_D00"
GH_seurat_batch_1@meta.data$condition[which(GH_seurat_batch_1@meta.data$condition=="CMO302")] <- "H1_D0"
GH_seurat_batch_1@meta.data$condition[which(GH_seurat_batch_1@meta.data$condition=="CMO303")] <- "H1_D7m"
GH_seurat_batch_1@meta.data$condition[which(GH_seurat_batch_1@meta.data$condition=="CMO304")] <- "H1_D14m"
GH_seurat_batch_1@meta.data$condition[which(GH_seurat_batch_1@meta.data$condition=="CMO305")] <- "H1_D7c"
GH_seurat_batch_1@meta.data$condition[which(GH_seurat_batch_1@meta.data$condition=="CMO306")] <- "H1_D14c"
GH_seurat_batch_1@meta.data$condition[which(GH_seurat_batch_1@meta.data$condition=="CMO307")] <- "H2_D00"
GH_seurat_batch_1@meta.data$condition[which(GH_seurat_batch_1@meta.data$condition=="CMO308")] <- "H2_D0"
GH_seurat_batch_1@meta.data$condition[which(GH_seurat_batch_1@meta.data$condition=="CMO309")] <- "H2_D7m"
GH_seurat_batch_1@meta.data$condition[which(GH_seurat_batch_1@meta.data$condition=="CMO310")] <- "H2_D14m"
GH_seurat_batch_1@meta.data$condition[which(GH_seurat_batch_1@meta.data$condition=="CMO311")] <- "H2_D7c"
GH_seurat_batch_1@meta.data$condition[which(GH_seurat_batch_1@meta.data$condition=="CMO312")] <- "H2_D14c"

GH_seurat_batch1_clean <- subset(GH_seurat_batch_1, subset = condition != "")
GH_seurat_batch1_clean_df <- GH_seurat_batch1_clean@meta.data

#Checking to make sure that we did not lose any cells in the process. The number of cells assigned to each CMO here should be the same as lines #87-98
value_counts <- table(GH_seurat_batch1_clean_df$condition)
print(value_counts)

save(GH_seurat_batch1_clean, file = "GH_seurat_batch1_clean.RData")
```

File for species assignment data. Species-specific assignments were determined using human-chimpanzee genome combination with ortho-exon annotation.
```{r}

spp <- readRDS("/project/gilad/ghousman/chondro-human-chimp/hc-chondro-time/chondro-time-evo/data/06_spp_assign.rds")

```

```{r load libraries, eval=FALSE, echo=TRUE}

library(Seurat)
library(dplyr)
library(stringi)
library(stringr)
library(ggplot2)
library(colorspace)
library(RColorBrewer)
library(tibble)

```


Adding species assignment to both batches and then will subset the seurat object to only include humans. The final objects are saved as GH_seurat_batch1_human_final_2024 and GH_seurat_batch2_human_final_2024 in project/gilad/emaan/time_project/data/processing_and_QC. These objects contain only human cells (no mulitplets). They do not have any cutoffs implemented and the mt data has to be added to the metadata. Note the object will appear as GH_seurat_batch1_sub and GH_seurat_batch2_sub. 
```{r}
load("/project/gilad/emaan/time_project/data/GH_seurat_batch1_clean.RData")

GH_seurat_batch1_clean_df <- GH_seurat_batch1_clean@meta.data

#Converting the actual first row into a column 
GH_seurat_batch1_clean_df<- rownames_to_column(GH_seurat_batch1_clean_df, var = "barcode")

#Getting rid of the -1 in the barcodes column
GH_seurat_batch1_clean_df$barcode <- gsub("-1", "", GH_seurat_batch1_clean_df$barcode)

#Subsetting spp table so it only has batch1 information
spp_batch1 <- spp[spp$batch == 'batch1', ]

library(dplyr)

merged_df <- left_join(GH_seurat_batch1_clean_df, spp_batch1, by = "barcode")
GH_seurat_batch1_clean_df$species <- merged_df$call

species_assignment <- GH_seurat_batch1_clean_df$species
names(species_assignment) = GH_seurat_batch1_clean_df$barcode

GH_seurat_batch1_clean <- AddMetaData(object=GH_seurat_batch1_clean, metadata = species_assignment, col.name ='species')

#Remove any multiplet values and NA values. And then subset to only human cells. 
GH_seurat_batch1_sub<- subset (GH_seurat_batch1_clean, subset = species != "Multiplet")
GH_seurat_batch1_sub <- subset(GH_seurat_batch1_clean, subset = species != "")
GH_seurat_batch1_sub <- subset(GH_seurat_batch1_clean, subset = species == "Human")

GH_seurat_batch1_sub_df <- GH_seurat_batch1_sub@meta.data
GH_seurat_batch1_sub_df <- GH_seurat_batch1_sub_df %>%
  mutate(
    full_condition = condition,
    condition = sub("^[^_]*_", "", condition),
    donor_ID = sub("_.*", "", full_condition)
  ) %>%
  select(orig.ident, nCount_RNA, nFeature_RNA, species, full_condition, condition, donor_ID)

GH_seurat_batch1_sub <- AddMetaData(GH_seurat_batch1_sub, GH_seurat_batch1_sub_df)
save(GH_seurat_batch1_sub, file = "GH_seurat_batch1_human_final_2024.RData")


#**********************************************************************************************
load("/project/gilad/emaan/time_project/data/processing_and_QC/GH_seurat_batch2_clean.RData")

GH_seurat_batch2_clean_df <- GH_seurat_batch2_clean@meta.data

#Converting the actual first row into a column 
GH_seurat_batch2_clean_df<- rownames_to_column(GH_seurat_batch2_clean_df, var = "barcode")

#Getting rid of the -1 in the barcodes column
GH_seurat_batch2_clean_df$barcode <- gsub("-1", "", GH_seurat_batch2_clean_df$barcode)

#Subsetting spp table so it only has batch1 information
spp_batch2 <- spp[spp$batch == 'batch2', ]

merged_df <- left_join(GH_seurat_batch2_clean_df, spp_batch2, by = "barcode")
GH_seurat_batch2_clean_df$species <- merged_df$call

species_assignment <- GH_seurat_batch2_clean_df$species
names(species_assignment) = GH_seurat_batch2_clean_df$barcode

GH_seurat_batch2_clean <- AddMetaData(object=GH_seurat_batch2_clean, metadata = species_assignment, col.name ='species')

#Remove any multiplet values and NA values. And then subset to only human cells. 
GH_seurat_batch2_sub<- subset (GH_seurat_batch2_clean, subset = species != "Multiplet")
GH_seurat_batch2_sub <- subset(GH_seurat_batch2_clean, subset = species != "")
GH_seurat_batch2_sub <- subset(GH_seurat_batch2_clean, subset = species == "Human")

GH_seurat_batch2_sub_df <- GH_seurat_batch2_sub@meta.data
GH_seurat_batch2_sub_df <- GH_seurat_batch2_sub_df %>%
  mutate(
    full_condition = condition,
    condition = sub("^[^_]*_", "", condition),
    donor_ID = sub("_.*", "", full_condition)
  ) %>%
  select(orig.ident, nCount_RNA, nFeature_RNA, full_condition, condition, donor_id)

GH_seurat_batch2_sub <- AddMetaData(GH_seurat_batch2_sub, GH_seurat_batch2_sub_df)

save(GH_seurat_batch2_sub, file = "GH_seurat_batch2_human_final_2024.RData")

```

QC Metrics: determining the nFeature RNA range and percent mitochondria DNA. Saved seurat object with % mitochondrial DNA included and nFeature RNA cutoffs.Final objects saved as GH_batch1_QC and GH_batch2_QC in /project/gilad/emaan/time_project/data 
```{r}
GH_seurat_batch1_sub[["percent.mt"]] <- PercentageFeatureSet(GH_seurat_batch1_sub, pattern = "^MT-")
GH_seurat_batch2_sub[["percent.mt"]] <- PercentageFeatureSet(GH_seurat_batch2_sub, pattern = "^MT-")

hist(GH_seurat_batch1_sub$nFeature_RNA)
hist(GH_seurat_batch1_sub$nCount_RNA)
hist(GH_seurat_batch1_sub$percent.mt)

mt_cutoff <- c(1,2,3,4,5,6,7,8,9,10, 15, 20, 25, 30, 35, 40, 45)
mt_count <- c()
for (i in mt_cutoff) {
  mt_count <- c(mt_count, sum(GH_seurat_batch1_sub@meta.data$percent.mt <= i))
}
mt_table <- data.frame(mt_cutoff, mt_count)
plot(mt_cutoff, mt_count, main="mt cutoff")

mt_cutoff <- c(1,2,3,4,5,6,7,8,9,10, 15, 20, 25, 30, 35, 40, 45)
mt_count <- c()
for (i in mt_cutoff) {
  mt_count <- c(mt_count, sum(GH_seurat_batch2_sub@meta.data$percent.mt <= i))
}
mt_table <- data.frame(mt_cutoff, mt_count)
plot(mt_cutoff, mt_count, main="mt cutoff")

# nFeature RNA
rna_cutoff <- c(100,200, 300, 400, 500, 600, 800, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000)
rna_count <- c()
for (i in rna_cutoff) {
  rna_count <- c(rna_count, sum(GH_seurat_batch1_sub@meta.data$nFeature_RNA >= i))
}
rna_table <- data.frame(rna_cutoff, rna_count)
plot(rna_cutoff, rna_count, main="nFeature RNA")

rna_cutoff <- c(100,200, 300, 400, 500, 600, 800, 1000, 2000, 3000, 4000, 5000, 6000, 7000, 8000)
rna_count <- c()
for (i in rna_cutoff) {
  rna_count <- c(rna_count, sum(GH_seurat_batch2_sub@meta.data$nFeature_RNA >= i))
}
rna_table <- data.frame(rna_cutoff, rna_count)
plot(rna_cutoff, rna_count, main="nFeature RNA")

#Violin plots for cutoff
VlnPlot(GH_seurat_batch1_sub, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
VlnPlot(GH_seurat_batch2_sub, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

#Adding the cutoffs in the seurat object
GH_batch1_QC<- subset(GH_seurat_batch1_sub, subset = nFeature_RNA > 200 & nFeature_RNA < 6000 & percent.mt < 5)
save(GH_batch1_QC, file = "GH_batch1_QC.RData")

GH_batch2_QC<- subset(GH_seurat_batch2_sub, subset = nFeature_RNA > 200 & nFeature_RNA < 6000 & percent.mt < 5)
save(GH_batch2_QC, file = "GH_batch2_QC.RData")

metadata <- GH_batch1_QC@meta.data

#Violin plot for nFeature RNA
ggplot(metadata, aes(x = factor(condition, levels = c("D00", "D0", "D7m", "D7c", "D14m", "D14c")), y = nFeature_RNA)) +
  labs(y = "UMIs per cell", x = "Condition") + 
  geom_violin()

#Violin plot for nCount RNA
ggplot(metadata, aes(x = factor(condition, levels = c("D00", "D0", "D7m", "D7c", "D14m", "D14c")), y = nCount_RNA)) +
  labs(y = "Genes per cell", x = "Condition") + 
  geom_violin()

#Violin plot for # of cells
condition_summary <- data.frame(condition = names(table(metadata$condition)),
                                count = as.vector(table(metadata$condition)))

ggplot(condition_summary, aes(x = factor(condition, levels = c("D00", "D0", "D7m", "D7c", "D14m", "D14c")), y=count)) + labs(x = "Condition", y = "Count") + geom_bar(stat = "identity", fill = "skyblue") 


#______________________________________________________________________________________________________________
#Violin plot for nFeature RNA
ggplot(metadata, aes(x = factor(condition, levels = c("D00", "D0", "D7m", "D7c", "D14m", "D14c")), y = nFeature_RNA)) +
  labs(y = "UMIs per cell", x = "Condition") + 
  geom_violin()

#Violin plot for nCount RNA
ggplot(metadata, aes(x = factor(condition, levels = c("D00", "D0", "D7m", "D7c", "D14m", "D14c")), y = nCount_RNA)) +
  labs(y = "Genes per cell", x = "Condition") + 
  geom_violin()

#Violin plot for # of cells
condition_summary <- data.frame(condition = names(table(metadata$condition)),
                                count = as.vector(table(metadata$condition)))

ggplot(condition_summary, aes(x = factor(condition, levels = c("D00", "D0", "D7m", "D7c", "D14m", "D14c")), y=count)) + labs(x = "Condition", y = "Count") + geom_bar(stat = "identity", fill = "skyblue") 
```

Creating QC figures for the combined EM+GH data. 
```{r}

load("/project/gilad/emaan/time_project/data/GH_seurat_batch1_human_final_2024.RData")
load("/project/gilad/emaan/time_project/data/GH_seurat_batch2_human_final_2024.RData")
load("/project/gilad/emaan/time_project/data/seurat_final_2023.RData")

GH_seurat_batch1_sub[["percent.mt"]] <- PercentageFeatureSet(GH_seurat_batch1_sub, pattern = "^MT-")
GH_seurat_batch2_sub[["percent.mt"]] <- PercentageFeatureSet(GH_seurat_batch2_sub, pattern = "^MT-")
seurat_final_2023[["percent.mt"]] <- PercentageFeatureSet(seurat_final_2023, pattern = "^MT-")

metadata_EM <- seurat_final_2023@meta.data
metadata_EM$batch <- "YRI"
metadata_EM_sub <- metadata_EM[, c("condition", "nFeature_RNA","nCount_RNA","percent.mt","batch")]

metadata_GH1 <- GH_seurat_batch1_sub@meta.data
metadata_GH1$batch <- "Human Panel"
metadata_GH1_sub <- metadata_GH1[, c("condition", "nFeature_RNA","nCount_RNA","percent.mt","batch")]

metadata_GH2 <- GH_seurat_batch2_sub@meta.data
metadata_GH2$batch <- "Human Panel"
metadata_GH2_sub <- metadata_GH2[, c("condition", "nFeature_RNA","nCount_RNA","percent.mt","batch")]

# Combine the data tables into one
combined_data <- rbind(metadata_EM_sub, metadata_GH1_sub, metadata_GH2_sub)

ggplot(combined_data, aes(x = factor(condition, levels = c("D00", "D0", "D7c", "D14c", "D7m", "D14m")), 
                          y = nFeature_RNA, 
                          fill = condition, 
                          alpha = batch)) +
  geom_violin(position = position_dodge(width = 0.9)) +
  scale_fill_manual(values = c("D00" = rgb(186, 85, 211, maxColorValue=255), "D0" = rgb(218, 75, 180, maxColorValue = 255), "D7m" = rgb(51, 153, 255, maxColorValue = 255), "D7c" = rgb(253, 93, 93, maxColorValue = 255), "D14m" = rgb(0, 102, 204, maxColorValue = 255), "D14c"=rgb(254, 192, 0, maxColorValue = 255))) +
  scale_alpha_manual(values = c("YRI" = 1, "Human Panel" = 0.3)) +
  labs(x = "Timepoint", y = "nFeature_RNA") +
  theme_classic()

ggplot(combined_data, aes(x = factor(condition, levels = c("D00", "D0", "D7c", "D14c", "D7m", "D14m")), 
                          y = nCount_RNA, 
                          fill = condition, 
                          alpha = batch)) +
  geom_violin(position = position_dodge(width = 0.9)) +
  scale_fill_manual(values = c("D00" = rgb(186, 85, 211, maxColorValue=255), "D0" = rgb(218, 75, 180, maxColorValue = 255), "D7m" = rgb(51, 153, 255, maxColorValue = 255), "D7c" = rgb(253, 93, 93, maxColorValue = 255), "D14m" = rgb(0, 102, 204, maxColorValue = 255), "D14c"=rgb(254, 192, 0, maxColorValue = 255))) +
  scale_alpha_manual(values = c("YRI" = 1, "Human Panel" = 0.3)) +
  labs(x = "Timepoint", y = "nCount_RNA") +
  theme_classic()

ggplot(combined_data, aes(x = factor(condition, levels = c("D00", "D0", "D7c", "D14c", "D7m", "D14m")), 
                          y = percent.mt, 
                          fill = condition, 
                          alpha = batch)) +
  geom_violin(position = position_dodge(width = 0.9)) +
  scale_fill_manual(values = c("D00" = rgb(186, 85, 211, maxColorValue=255), "D0" = rgb(218, 75, 180, maxColorValue = 255), "D7m" = rgb(51, 153, 255, maxColorValue = 255), "D7c" = rgb(253, 93, 93, maxColorValue = 255), "D14m" = rgb(0, 102, 204, maxColorValue = 255), "D14c"=rgb(254, 192, 0, maxColorValue = 255))) +
  scale_alpha_manual(values = c("YRI" = 1, "Human Panel" = 0.3)) +
  labs(x = "Timepoint", y = "percent.mt") +
  theme_classic()

condition_summary_EM <- data.frame(condition = names(table(metadata_EM_sub$condition)),
                                count = as.vector(table(metadata_EM_sub$condition)))

condition_summary_GH1<- data.frame(condition = names(table(metadata_GH1_sub$condition)),
                                count = as.vector(table(metadata_GH1_sub$condition)))

condition_summary_GH2<- data.frame(condition = names(table(metadata_GH2_sub$condition)),
                                count = as.vector(table(metadata_GH2_sub$condition)))

condition_summary_EM$batch <- "YRI"
condition_summary_GH1$batch <- "Human Panel"
condition_summary_GH2$batch <- "Human Panel"

combined_summary <- rbind(condition_summary_EM, condition_summary_GH1, condition_summary_GH2)

ggplot(combined_summary, aes(x = factor(condition, levels = c("D00", "D0", "D7c", "D14c", "D7m", "D14m")), 
                          y = count, 
                          fill = condition, 
                          alpha = batch)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("D00" = rgb(186, 85, 211, maxColorValue=255), "D0" = rgb(218, 75, 180, maxColorValue = 255), "D7m" = rgb(51, 153, 255, maxColorValue = 255), "D7c" = rgb(253, 93, 93, maxColorValue = 255), "D14m" = rgb(0, 102, 204, maxColorValue = 255), "D14c"=rgb(254, 192, 0, maxColorValue = 255))) +
  scale_alpha_manual(values = c("YRI" = 1, "Human Panel" = 0.3)) +
  labs(x = "Timepoint", y = "count") +
  theme_classic()


```

Integrating EM+GH data and creating UMAPs. Files saved in /project/gilad/emaan/time_project/data/integrating_EM_GH. The file name format follows integrate_all_human_xpca_nonreg.Rdata or integrate_all_human_xpca_reg.RData. 
````{r}

load("/project/gilad/emaan/time_project/data/GH_batch2_QC.RData")
load("/project/gilad/emaan/time_project/data/GH_batch1_QC.RData")
load("/project/gilad/emaan/time_project/data/seurat_QC_2024.RData")

#Seurat objects being used
NA18856 <- subset(seurat_QC_2024, subset = donor_id == 'NA18856')
NA18855 <- subset(seurat_QC_2024, subset = donor_id == 'NA18855')
NA19160 <- subset(seurat_QC_2024, subset = donor_id == 'NA19160')
H1 <- subset(GH_batch1_QC, subset = donor_ID == 'H1')
H2 <- subset(GH_batch1_QC, subset = donor_ID == 'H2')
H3 <- subset(GH_batch2_QC, subset = donor_ID == 'H3')
H5 <- subset(GH_batch2_QC, subset = donor_ID == 'H5')

#Creating a list of the Seurat objects
individual_list <- list(NA18856, NA18855, NA19160, H1,H2,H3,H5)

# normalize and identify variable features for each dataset independently
individual_list <- lapply(X = individual_list, FUN = function(x) {
    x <- NormalizeData(x)
    x <- FindVariableFeatures(x, selection.method = "vst", nfeatures = 2000)
})

#Select features that are repeatedly variable across data sets for integration
features <- SelectIntegrationFeatures(object.list = individual_list)
cell.anchors <- FindIntegrationAnchors(object.list = individual_list, anchor.features = features)

#Creating the integrated data set
cells.combined <- IntegrateData(anchorset = cell.anchors)
DefaultAssay(cells.combined) <- "integrated"

#Standard workflow for visualization and clustering 
cells.combined <- ScaleData(cells.combined, verbose = FALSE)
#If you want to regress out any variables
#cells.combined <- ScaleData(cells.combined, vars.to.regress=c("nCount_RNA","percent.mt"))

cells.combined <- RunPCA(cells.combined, npcs = 50, verbose = FALSE)

#Trying to find how many PCs to include
ElbowPlot(cells.combined)
DimHeatmap(cells.combined, dims = 1:50, cells = 500, balanced = TRUE)
#cells.combined at this step saved as cells.combined_integrated_sw in /project/gilad/emaan/time_project/data

cells.combined <- RunUMAP(cells.combined, reduction = "pca", dims = 1:30)
cells.combined <- FindNeighbors(cells.combined, reduction = "pca", dims = 1:30)

#Trying different resolutions. 0.5 looks the best. Note: resolution greater than 1 results in larger number of clusters, resolution smaller than 1 results in smaller number of clusters

cells.combined_1 <- FindClusters(cells.combined, resolution = 1)
cells.combined_1.2 <- FindClusters(cells.combined, resolution = 1.2)
cells.combined_0.4 <- FindClusters(cells.combined, resolution = 0.4)
cells.combined <- FindClusters(cells.combined, resolution = 0.5)

# Visualization
p1 <- DimPlot(cells.combined, reduction = "umap", group.by = "condition")
p2 <- DimPlot(cells.combined, reduction = "umap", label = TRUE, repel = TRUE)
p1 + p2

DimPlot(cells.combined_1, reduction = "umap", split.by = "condition",label=TRUE)
DimPlot(cells.combined_1.2, reduction = "umap", split.by = "condition",label=TRUE)
DimPlot(cells.combined_0.4, reduction = "umap", split.by = "condition",label=TRUE)
DimPlot(cells.combined, reduction = "umap", split.by = "condition",label=TRUE)

DefaultAssay(cells.combined_0.4) <- "RNA"

Idents(cells.combined_0.4) <- factor(Idents(cells.combined_0.4))
markers.to.plot <- c("SOX9", "SOX5", "SOX6", "COL2A1", "ACAN", "COL9A1", "MATN3", "SPARC", "COL11A1","PRG4")
DotPlot(cells.combined_0.4, features = markers.to.plot, cols = c("lightgrey", "blue"), dot.scale = 8, col.min = 0) + RotatedAxis()

plots <- VlnPlot(cells.combined_0.5, features = c("SOX5"), split.by = "condition",
    pt.size = 0, combine = FALSE)
wrap_plots(plots = plots, ncol = 1)

save(cells.combined, file = "/project/gilad/emaan/time_project/data/integrating_EM_GH/integrate_all_human_30pca_nonreg.RData")

save(cells.combined, file = "/project/gilad/emaan/time_project/data/integrating_EM_GH/integrate_all_human_15pca_nonreg.RData")


#Checking pearson correlation before and after regression
FeatureScatter(cells.combined, feature1 = "nCount_RNA", feature2 = "PC_1")
FeatureScatter(cells.combined, feature1 = "percent.mt", feature2 = "PC_1")



```

Integrating EM+GH data using GH code. The code used is in /project/gilad/emaan/time_project/code/mapping_all_human_GHcode.sh. The output files are saved in /project/gilad/emaan/time_project/data/integrating_EM_GH. Similar code was used to integrate EM+GH+Wu data using GH code. The code is in /project/gilad/emaan/time_project/code/mapping_wu_GHcode.sh. The file is saved in integrate_Wu_var_cca_reg.RData. 
```{r}

#visualizing integrated dataset. 
DimPlot(integrate,
        group.by="condition",
        label=TRUE,
        label.size=2,
        reduction="umap") +
  labs(x="UMAP1",y="UMAP2") +
  NoLegend()

# UMAP plot without labels
umap_plot <- DimPlot(integrate, group.by = "condition", reduction = "umap", label=FALSE, label.size=5) +
  labs(x = "UMAP1", y = "UMAP2") +
  NoLegend()

#Checking pearson correlation before and after regression to make sure it worked
FeatureScatter(integrate, feature1 = "nCount_RNA", feature2 = "PC_1")
FeatureScatter(integrate, feature1 = "percent.mt", feature2 = "PC_1")
```



